File: Leetcode/Two Sum/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int length = nums.length;
        int[] result = new int[2];
        int num1;
        int num2;
        for (int i = 0; i < length; i++){
            num1 = nums[i];
            for (int j = 1; j < length + 1; j++){
                if (j == length){
                    //If index j reaches length of the array, it wraps around and becomes 0.
                    j = 0; 
                }
                if (i == j){
                    //Indices i and j can never be at the same place in the array.
                    break;
                }
                num2 = nums[j];
                if ((num1 + num2) == target){
                    //If the sum of num1 and num2 is the target number, assign its indices i and j to the result array.
                    result[0] = i;
                    result[1] = j;
                    return result;
                }
            }
        }
        return result;
    }
}

File: Leetcode/Two Sum/TwoSum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        length = len(nums)
        result = []
        i = 0
        while i < length:
            num1 = nums[i]
            j = i + 1
            while j < length + 1:
                if (j == length):
                    j = 0
                if (i == j):
                    break
                num2 = nums[j]
                if ((num1 + num2) == target):
                    result.append(i)
                    result.append(j)
                    return result
                j += 1
            i += 1
        return result


                
                
        